DIP.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from typing import NamedTuple, List, Any, Dict
+from pytest import mark
+
+#### TESTS
+
```

Comment:
Since there's already `test_attendance_worflows.py`, Maybe it would be good to move this test to that file.